modin/engines/base/io/text/text_file_dispatcher.py

```diff
@@ -232,29 +286,165 @@ def _read_rows(
+
+                def stepper():
+                    row_number = 0
+                    index_to_compare = 0
+                    while index_to_compare < len(skiprows):
+                        if skiprows[index_to_compare] == row_number:
+                            index_to_compare += 1
+                            yield 0
+                        else:
+                            yield 1
+                        row_number += 1
+                    while True:
+                        yield 1
+
+            else:
+
+                def stepper():
+                    while True:
+                        yield 1
+
+            return stepper()
+
+        if should_handle_skiprows:
+            skiprows_handler = skiprows_handler_builder(skiprows)
 
         for line in f:
             if is_quoting and line.count(quotechar) % 2:
                 outside_quotes = not outside_quotes
             if outside_quotes:
-                rows_read += 1
-                if rows_read >= nrows:
+                if should_handle_skiprows:
+                    rows_considered += next(skiprows_handler)
+                else:
+                    rows_considered += 1
+                rows_readed += 1
+                if stop_condition(rows_readed=rows_considered):
                     break
-
         # case when EOF
         if not outside_quotes:
-            rows_read += 1
+            rows_readed += 1
+
+        if save_position:
+            f.seek(start_position, os.SEEK_SET)
+
+        return outside_quotes, rows_readed, rows_considered
+
+    @classmethod
+    def handle_skiprows(
+        cls,
+        skiprows: Union[Sequence[int], Callable, None],
+        start_row: int,
+        extra_skiprows: Union[Sequence, int, None] = None,
+    ):
+        """
+        Handle skiprows parameter according to the reference start_row.
+
+        Parameters
+        ----------
+        skiprows: array or callable
+            skiprows object that should be aligned to the start_row.
+        start_row: int
+            skiprows alignement reference.
+        extra_skiprows: array or int
+            Additional rows numbers that should be skipped (indexed
+            from start_row).
+
+        Returns
+        -------
+        Skiprows object, aligned to the start_row.
+        """
+        if extra_skiprows is None:
+            extra_skiprows = []
+        elif not isinstance(extra_skiprows, (list, tuple)):
+            extra_skiprows = [extra_skiprows]
+
+        def skiprows_wrapper(n):
+            return n in extra_skiprows or skiprows(n + start_row)
+
+        if callable(skiprows):
+            new_skiprows = skiprows_wrapper
+        elif is_list_like(skiprows):
+            start = np.searchsorted(skiprows, start_row)
+            new_skiprows = np.concatenate(
+                [extra_skiprows, skiprows[start:] - start_row]
+            )
+            if len(extra_skiprows) > 0:
+                new_skiprows = np.sort(new_skiprows)
```

Comment:
why do we need to sort this list?

Reply:
In the case of `extra_skiprows` presence, `new_skiprows` can be unsorted array that can cause wrong behavior of `_read_rows` function.